src/core/thread/mle_router.cpp

```diff
@@ -337,6 +337,8 @@ void MleRouter::SetStateRouter(uint16_t aRloc16)
 
 void MleRouter::SetStateLeader(uint16_t aRloc16)
 {
+    IgnoreError(Get<MeshCoP::ActiveDataset>().Restore());
```

Comment:
Instead of using `IgnoreError`, I think we should return the error or log it in-place. thoughts?

Generally, I think we should not use `IgnoreError` in new code and try to remove existing `IgnoreError` where possible.

Reply:
I think this is ignored because the error case is not found only.

Reply:
Adding a log message in place of every place where we might otherwise use `IgnoreError()` will blow up the code size when logging is enabled. As @bukepo mentioned, adding a log message here is not useful because there is a similar log message that will tell us the same information in `ActiveDataset::GenerateLocal()` (which is also called in this path). Thoughts?

Reply:
You mean failed to restore operational dataset is okay in context of `SetStateLeader`?

Reply:
Yes, not having an active and/or pending operational dataset is acceptable in the current implementation. `ActiveDataset::GenerateLocal()` will generate one if no active operational dataset exists.

Reply:
I agree that this error can be ignored because the semantic seems to be `try to restore operational dataset`, in current context.